arrow/src/array/data.rs

```diff
@@ -936,68 +936,68 @@ impl ArrayData {
             )));
         }
 
+        self.validate_dictionary_offest()?;
+
+        // validate all children recursively
+        self.child_data
+            .iter()
+            .enumerate()
+            .try_for_each(|(i, child_data)| {
+                child_data.validate_full().map_err(|e| {
+                    ArrowError::InvalidArgumentError(format!(
+                        "{} child #{} invalid: {}",
+                        self.data_type, i, e
+                    ))
+                })
+            })?;
+
+        Ok(())
+    }
+
+    pub fn validate_dictionary_offest(&self) -> Result<()> {
```

Comment:
there is a typo here I think:

```suggestion
    pub fn validate_dictionary_offset(&self) -> Result<()> {
```

Reply:
It might be clearer to call this `validate_dictionary_offsets_full`